src/Ormolu/Diff/ParseResult.hs

```diff
@@ -92,14 +93,17 @@ matchIgnoringSrcSpans a = genericQuery a
                   `extQ` considerEqual @SourceText
                   `extQ` hsDocStringEq
                   `extQ` importDeclQualifiedStyleEq
-                  `extQ` unicodeArrowStyleEq
                   `extQ` considerEqual @(LayoutInfo GhcPs)
                   `extQ` classDeclCtxEq
                   `extQ` derivedTyClsParensEq
                   `extQ` considerEqual @EpAnnComments -- ~ XCGRHSs GhcPs
-                  `extQ` considerEqual @TokenLocation
+                  `extQ` considerEqual @TokenLocation -- in LHs(Uni)Token
                   `extQ` considerEqual @EpaLocation
                   `ext2Q` forLocated
+                  -- unicode-related
+                  `extQ` considerEqual @(HsUniToken "->" "→")
+                  `extQ` considerEqual @(HsUniToken "::" "∷")
```

Comment:
Ideally, we would ignore differences for [`HsUniToken tok utok`](https://hackage.haskell.org/package/ghc-lib-parser-9.4.4.20221225/docs/Language-Haskell-Syntax-Extension.html#t:HsUniToken) for all `Symbol`s `tok` and `utok`. This does not work via `ext2Q` as the `Data (HsUniToken tok utok)` instance requires `KnownSymbol` instead of `Data` for `tok` and `utok`, so a generic solution would instead look like this:
```diff
diff --git a/src/Ormolu/Diff/ParseResult.hs b/src/Ormolu/Diff/ParseResult.hs
index 249c36a..e4a8582 100644
--- a/src/Ormolu/Diff/ParseResult.hs
+++ b/src/Ormolu/Diff/ParseResult.hs
@@ -20,6 +20,7 @@ import GHC.Types.SrcLoc
 import Ormolu.Parser.CommentStream
 import Ormolu.Parser.Result
 import Ormolu.Utils
+import Type.Reflection qualified as TR
 
 -- | Result of comparing two 'ParseResult's.
 data ParseResultDiff
@@ -83,6 +84,9 @@ matchIgnoringSrcSpans a = genericQuery a
           if x' == (y' :: ByteString)
             then Same
             else Different []
+      | Just rep <- isHsUniToken x,
+        Just rep' <- isHsUniToken y =
+          if rep == rep' then Same else Different []
       | typeOf x == typeOf y,
         toConstr x == toConstr y =
           mconcat $
@@ -101,14 +105,18 @@ matchIgnoringSrcSpans a = genericQuery a
                   `extQ` considerEqual @EpaLocation
                   `ext2Q` forLocated
                   -- unicode-related
-                  `extQ` considerEqual @(HsUniToken "->" "→")
-                  `extQ` considerEqual @(HsUniToken "::" "∷")
                   `extQ` considerEqual @(HsLinearArrowTokens GhcPs)
               )
               x
               y
       | otherwise = Different []
 
+    isHsUniToken :: (Typeable a) => a -> Maybe TR.SomeTypeRep
+    isHsUniToken (typeOf -> rep)
+      | TR.SomeTypeRep (TR.App (TR.App t _) _) <- rep =
+          rep <$ TR.eqTypeRep t (TR.typeRep @HsUniToken)
+      | otherwise = Nothing
+
     considerEqualVia ::
       forall a.
       (Typeable a) =>
```
This is more complex, but would have the advantage that we don't have to add more cases here if `HsUniToken` is used for more many unicode variants in the future in the AST.

Reply:
:+1: Let's keep this in mind in case we discover more of these. For now a simpler fix will do.